crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -140,66 +204,85 @@ impl GitCommandProvider {
     }
 
     /// Open a repo, erroring if `path` is not a repo or is a subdirectory of a repo
-    //
-    // TODO should share more code with discover?
-    pub fn open<P: AsRef<Path>>(path: P) -> Result<Self, GitCommandOpenError> {
-        let out_str = GitCommandProvider::run_command(
-            GitCommandProvider::new_command(&Some(&path))
+    pub fn open_with<P: AsRef<Path>>(
```

Comment:
nitpick: I think `open_with_options` is a slightly clearer name

Reply:
Ping me if you don't plan on renaming and I'll merge.

Reply:
I dont plan for it now.
If its itching or get ambiguous in some commands i'll consider making it more explicit